File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    look_up := make(map[int]int)

    // for i, num := range nums {
    //     look_up[num] = i
    // }


    for i, num := range nums {
        compliment := target - num

        if idx, ok := look_up[compliment]; ok {
            return []int{idx, i}
        }

        look_up[num] = i
    }

    return []int{}
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> track;
        for (int i = 0; i < nums.size(); ++i) {
            auto num = nums[i];
            auto it = track.find(target - num);

            if (it != track.end() && it->second != i) {
                return {it->second, i};
            }
            track[num] = i;
        }

        return {};
    }
};